### ISSUE TITLE
no syntax highlighting

### ISSUE THREAD

**Issue by gour** [onivim/oni2#1933]
Hello,

today, after installing #5d000137 AppImage I noticed that Oni2 does nor do syntax highlighting for any filetype I tried. THought it's maybe problem with some extensions, but even after rm-ing all installed extension and opening simple *.html file, still it does not work.

Here is is the debug log file:
[oni2.log](https://github.com/onivim/oni2/files/4775276/oni2.log)

and here is is output of `--checkhealth` as suggested by Ryan:

[oni3.log](https://github.com/onivim/oni2/files/4775277/oni3.log)

I'm on Fedora fc32 Linux.

Sincerely,  
Gour

**Comment by bryphe** [comment]
Thanks for logging the issue and for the logs, @gour ! The health-check looks good (

Do you happen to have any custom configuration, or are using a custom theme or font? The configuration file would be at `~/.config/oni2/configuration.json`. 

Possibly related:
- #1836
- #1879

**Comment by gour** [comment]
> Do you happen to have any custom configuration, or are using a custom theme or font?

Nope, just a pristine install...

Also, strange thing is that Explore/Sidebar are not visible:

![oni2](https://user-images.githubusercontent.com/114592/84586857-13566300-ae1b-11ea-962c-1656ead2eca3.png)

**Comment by CrossR** [comment]
The sidebar/explorer not being shown is part of Zen mode (You can toggle it off in the Ctrl-Shift-P menu).

What file types are you missing syntax highlights in?  The original post says none, but the screenshot has it for JSON files at least.

**Comment by gour** [comment]
> The sidebar/explorer not being shown is part of Zen mode (You can toggle it off in the Ctrl-Shift-P menu).

Ahh, OK. I wondered why it was disabled by default..

> What file types are you missing syntax highlights in? 

E.g. HTML.

**Comment by bryphe** [comment]
Thanks for the update, @gour ! 

The HTML issue may be related to https://github.com/onivim/oni2/issues/1879 (a bug with our default theme, `OneDark Pro`... interestingly, other themes seem to work...)

**Comment by gour** [comment]
> The HTML issue may be related to #1879 (a bug with our default theme, `OneDark Pro`... interestingly, other themes seem to work...)

Yes, I can confirm that with few other themes I've tried, everything is fine. :smile:

**Comment by thismat** [comment]
The official Dracula theme is also having this issue. I'm adding what I've pin-pointed in the theme here for reference, maybe it will help in pin-pointing what's going on with theme compatibility.

- Javascript and Typescript were the primary issue, seems to vary based on theme.
- Markdown works fine
- HTML works fine

I didn't test more than those languages.

### Pin-Pointed Theme Setting

Dracula Theme yaml: https://github.com/dracula/visual-studio-code/blob/master/src/dracula.yml
Line: 410-411

The yaml compiles into this:

```json
// Line 280-287 dracula.json
{
  "scope": [
    "source"
  ],
  "settings": {
    "foreground": "#F8F8F2"
  }
}
```

Example of what is happening: 

![image](https://user-images.githubusercontent.com/95755/89555459-b2715880-d7d5-11ea-9cf3-70fab45a0975.png)

With block 280-287 removed:

![image](https://user-images.githubusercontent.com/95755/89555549-ccab3680-d7d5-11ea-91fc-5e2b489776e4.png)

### Additional Notes

I thought maybe it was an order of how it applies the theme styles that could possibly be causing an issue, maybe layering that source on top of other rules. I moved the source scoped block to the bottom of the theme and tried again, it did not work.

Initially I saw keywords "let this const var" etc, failing. There is an edge case in the theme at lines 1112-1120 (dracula.json):

```json
{
    "name": "Shell variables prefixed with \"$\" (edge case)",
    "scope": [
        "source.shell variable.other"
    ],
    "settings": {
        "foreground": "#BD93F9"
    }
},
```

Removing this solved that issue initially, but after I reverted the theme completely to a fresh install and just removed the "source" block, it didn't require I remove this one again. Just thought it was worth mentioning that behavior.

I hope this helps in narrowing down the issue.

**Comment by bryphe** [comment]
This should be fixed now with #2301 👍  LMK if you're still seeing issues with any theme / languages
